File: java/pca-model/src/main/java/com/sdl/web/pca/client/contentmodel/ContentComponent.java
package com.sdl.web.pca.client.contentmodel;

import com.sdl.web.pca.client.contentmodel.enums.ItemType;
import java.util.List;

/**
*Represents a component which has content.
*/
public interface ContentComponent {
																														

		String getCreationDate();
		void setCreationDate(String creationDate);

		CustomMetaConnection getCustomMetas();
		void setCustomMetas(CustomMetaConnection customMetas);

		String getId();
		void setId(String id);

		String getInitialPublishDate();
		void setInitialPublishDate(String initialPublishDate);

		int getItemId();
		void setItemId(int itemId);

		ItemType getItemType();
		void setItemType(ItemType itemType);

		String getLastPublishDate();
		void setLastPublishDate(String lastPublishDate);

		Integer getNamespaceId();
		void setNamespaceId(Integer namespaceId);

		Integer getOwningPublicationId();
		void setOwningPublicationId(Integer owningPublicationId);

		int getPublicationId();
		void setPublicationId(int publicationId);

		Integer getSchemaId();
		void setSchemaId(Integer schemaId);

		List<TaxonomyItem> getTaxonomies();
		void setTaxonomies(List<TaxonomyItem> taxonomies);

		String getTitle();
		void setTitle(String title);

		String getUpdatedDate();
		void setUpdatedDate(String updatedDate);

		boolean getMultiMedia();
		void setMultiMedia(boolean multiMedia);	
}

File: net/src/Sdl.Tridion.Api.Client/ContentModel/Patches/ContentComponent.cs
using System.Collections.Generic;

namespace Sdl.Tridion.Api.Client.ContentModel.Patches
{
    public class ContentComponent : IContentComponent
    {
        public string CreationDate { get; set; }
        public CustomMetaConnection CustomMetas { get; set; }
        public string Id { get; set; }
        public string InitialPublishDate { get; set; }
        public int ItemId { get; set; }
        public ItemType ItemType { get; set; }
        public string LastPublishDate { get; set; }
        public ContentNamespace NamespaceId { get; set; }
        public int? OwningPublicationId { get; set; }
        public int PublicationId { get; set; }
        public int? SchemaId { get; set; }
        public List<ITaxonomyItem> Taxonomies { get; set; }
        public string Title { get; set; }
        public string UpdatedDate { get; set; }
        public bool? MultiMedia { get; set; }
    }
}
